src/sql.rs

```diff
@@ -0,0 +1,53 @@
+use jobs;
+use mysql::{OptsBuilder, Params, Pool};
+use serde_json::{json, Number};
+use std::error::Error;
+use std::io::Result;
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+
+lazy_static! {
+    static ref POOL: Arc<Mutex<Option<Pool>>> = Arc::new(Mutex::new(None));
+}
+
+// helper functions to prevent uglification
+fn err_to_json(e: &dyn Error) -> String {
+    json!({
+        "status": "err",
+        "data": &e.to_string()
+    })
+    .to_string()
+}
+
+fn sql_connect_pool(
+    host: &str,
+    port: u16,
+    user: &str,
+    pass: &str,
+    db: &str,
+    timeout: Duration,
+    max_threads: usize,
+) -> Result<String> {
```

Comment:
I should probably turn this into `Result<String, Box<dyn Error>>` and just let the byond functions handle translating errors into JSON, so I can use `?` and for less `err_to_json` spam